src/main/java/bio/terra/service/profile/ProfileService.java

```diff
@@ -212,4 +225,47 @@ public void verifyAccount(BillingProfileRequestModel request, AuthenticatedUserR
         }
     }
 
+    // -- profile upgrade --
+
+    // Billing profiles prior to the introduction of this service did not have sam resources behind them.
+    // This code generates sam resources for any billing profile it does not have one. It sets the
+    // stewards group as the owner of the billing profile.
+    //
+    // We assume this is being run as a Steward and that the Steward has access to any billing
+    public UpgradeResponseModel upgradeProfileResources(UpgradeModel request, AuthenticatedUserRequest user) {
+        Instant startTime = Instant.now();
+        List<BillingProfileModel> profiles = profileDao.getOldBillingProfiles();
+        List<UUID> resources = iamService.listAuthorizedResources(user, IamResourceType.SPEND_PROFILE);
+        Set<String> profileResourceSet;
+        if (resources == null) {
+            profileResourceSet = Collections.EMPTY_SET;
+        } else {
+            profileResourceSet = resources.stream().map(r -> toString()).collect(Collectors.toSet());
+        }
+
+        for (BillingProfileModel profile : profiles) {
+            String profileId = profile.getId();
+            if (!profileResourceSet.contains(profileId)) {
+                try {
+                    // No Sam profile, so let's make one and make stewards owner
+                    logger.info("Creating profile resource for {} ({})", profile.getProfileName(), profileId);
+                    iamService.createProfileResource(user, profile.getId());
+                    logger.info("Adding steward owners for {} ({})", profile.getProfileName(), profileId);
+                    iamService.addPolicyMember(
+                        user,
+                        IamResourceType.SPEND_PROFILE,
+                        UUID.fromString(profileId),
+                        "owner", //IamRole.OWNER.name(),
```

Comment:
Any reason not to use the constant?  Looks like the toString lowercases it...so `IamRole.OWNER.toString()` would do the trick

Reply:
I should add a comment. There is a bug with sam where it is being case sensitive on this particular call.